flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvFormatITCase.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.csv;
+
+import org.apache.flink.formats.common.TimeFormats;
+import org.apache.flink.table.planner.runtime.utils.TestData;
+import org.apache.flink.table.planner.utils.JavaScalaConversionUtil;
+import org.apache.flink.table.planner.utils.JsonPlanTestBase;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.utils.DateTimeUtils.unixTimestampToLocalDateTime;
+
+/** Tests for the CSV file format. */
+public class CsvFormatITCase extends JsonPlanTestBase {
+
+    @Test
+    public void testProjectPushDown() throws Exception {
```

Comment:
Here there are no DataStream tests whatsoever. Please add them to cover your new functionality (don't bother covering old stuff).

Reply:
Do you mean testing the new `CsvBulkWriter` and `CsvReaderFormat` not anything related to the ser/de schema-based (old) stuff? Another question - you point at the projection pushdown test, not sure if you meant it explicitly (new functionality?) or if it is a coincidence. Do we even have facilities for communicating pushdown from the streaming API?

Reply:
> Do you mean testing the new CsvBulkWriter + CsvReaderFormat and not anything related to the ser/de schema-based (old) stuff?

Yes

> Another question - you point at the projection pushdown test, not sure if you meant it explicitly (new functionality?) or if it is a coincidence. Do we even have facilities for communicating pushdown from the streaming API?

I can only write comments on changed lines. So it's a coincidence.